File: go/internal/agent/pi.go
package agent

import (
	"context"
	"fmt"
	"time"
)

type PiAgent struct {
	BaseAgent
}

func NewPiAgent() *PiAgent {
	return &PiAgent{}
}

func (a *PiAgent) ScanMonorepoWorkspaces(ctx context.Context, directory string) ([]string, error) {
	fmt.Printf("Scanning monorepo workspaces in %s...\n", directory)
	time.Sleep(200 * time.Millisecond)
	return []string{"packages/core", "packages/web", "apps/desktop"}, nil
}

File: rust/src/pi.rs
use std::time::Duration;
use tokio::time::sleep;

pub struct PiAgent {}

impl PiAgent {
    pub fn new() -> Self {
        Self {}
    }

    pub async fn scan_monorepo_workspaces(&self, directory: &str) -> Result<Vec<String>, String> {
        println!("Scanning monorepo workspaces in {}...", directory);
        sleep(Duration::from_millis(200)).await;
        Ok(vec![
            "packages/core".to_string(),
            "packages/web".to_string(),
            "apps/desktop".to_string(),
        ])
    }
}
